src/plugins/autoColumnSize/autoColumnSize.js

```diff
@@ -204,7 +211,7 @@ class AutoColumnSize extends BasePlugin {
     const rowsRange = typeof rowRange === 'number' ? { from: rowRange, to: rowRange } : rowRange;
 
     rangeEach(columnsRange.from, columnsRange.to, (col) => {
-      if (force || (this.widths[col] === void 0 && !this.hot._getColWidthFromSettings(col))) {
+      if (force || (this.columnWidthsMap.getValueAtIndex(col) === void 0 && !this.hot._getColWidthFromSettings(col))) {
```

Comment:
Same as below. I don't know if you should fix bugs within this task.

Reply:
Sure, small bugs if you see one. The goal is to fix as many issues with indexes as possible.

Reply:
You're right. I recognised a similar change as a potential bug when I worked on #6173.

Reply:
What about this part of the code?